r/vignettes/developers/install_details.Rmd

```diff
@@ -0,0 +1,90 @@
+# How the R package is installed
```

Comment:
I don't actually want to call this "all the gory details of how the R package (could) be installed" but we probably should either make the title something like that or make the first paragraph a disclaimer of "none of this should be necessary knowledge to use Arrow, it's here as documentation and explanation in case you're developing or something goes terribly wrong